solidity/contracts/middleware/V2CompatibilityRouter.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import {Router} from "../Router.sol";
+import {TypeCasts} from "../libs/TypeCasts.sol";
+import {IMessageRecipient} from "../../interfaces/IMessageRecipient.sol";
+
+/*
+ * @title The Hello World App
+ * @dev You can use this simple app as a starting point for your own application.
+ */
+contract V2CompatibilityRouter is Router {
+    mapping(uint32 => uint32) v1ToV2Domain;
+    mapping(uint32 => uint32) v2ToV1Domain;
+
+    function initialize(
+        address _owner,
+        address _abacusConnectionManager,
+        address _interchainGasPaymaster
+    ) public initializer {
+        // Transfer ownership of the contract to deployer
+        _transferOwnership(_owner);
+        // Set the addresses for the ACM and IGP
+        // Alternatively, this could be done later in an initialize method
+        _setAbacusConnectionManager(_abacusConnectionManager);
+        _setInterchainGasPaymaster(_interchainGasPaymaster);
+    }
+
+    /**
+     * @notice Adds a domain ID mapping from v1/v2 domain IDs and vice versa
+     * @param _v1Domains An array of v1 domain IDs
+     * @param _v2Domains An array of v2 domain IDs
+     */
+    function mapDomains(
+        uint32[] calldata _v1Domains,
+        uint32[] calldata _v2Domains
+    ) external onlyOwner {
+        for (uint256 i = 0; i < _v1Domains.length; i += 1) {
+            v1ToV2Domain[_v1Domains[i]] = _v2Domains[i];
```

Comment:
think it's worth any input validation? e.g. making sure the mapping isn't already set or something

Reply:
Yeah I was thinking about it, but felt like i wanted to just maintain full flexibility for the owner